Improve RSS entries for statuses

ClearlyClaire opened the pull request:
- Render polls in both accounts and tags serializers
- Refactor RSS serializers
- Change title preview to include ellipsis when truncated
- Change title preview to show CW instead of toot text
- Add tests

ClearlyClaire committed:
Improve RSS entries for statuses

- Render polls in both accounts and tags serializers
- Refactor RSS serializers
- Change title preview to include ellipsis when truncated
- Change title preview to show CW instead of toot text
- Add tests

Gargron commented on app/models/status.rb:
```diff
@@ -198,10 +198,20 @@ def preview_card
+    preview = proper.spoiler_text.presence || proper.text
+    if preview.length > 30 || preview[0, 30].include?("\n")
+      preview = preview[0, 30]
+      preview = preview[0, preview.index("\n").presence || 30] + '…'
```
The fact that we're re-implementing ActiveSupport text helpers in here makes me think maybe this method has overstayed its welcome here and maybe it belongs in some presentation layer.

ThibG replied:
Yeah, I thought about moving it to the RSS serializers at first but forgot it was used by the OEmbed endpoint. I'll think about where to move it (maybe the Formatter)

Gargron replied:
I'm not sure why the OEmbed endpoint needs titles, to be honest. Twitter's doesn't give tweets titles. We could get rid of it.

ThibG replied:
Does twitter use OEmbed?

Gargron replied:
Yeah!

ClearlyClaire committed:
Remove title from OEmbed serialization

Twitter doesn't serialize title either, and tihs allows us to move the
title formatting code to the RSS serializers.
